### ISSUE TITLE
amp-bind-macro sample code documentation is incomplete

### ISSUE THREAD

**Issue by philkrie** [ampproject/amp.dev#3301]
### Current Behavior
https://amp.dev/documentation/components/amp-bind/?format=websites#defining-macros-with-amp-bind-macro

amp-bind-macro shown to have an arguments attribute. It should actually be 'args'. AMP isn't throwing an error, so this is probably very confusing.

Example here: https://amp-snippets.glitch.me/amp-bind-macro-issue.html

**Comment by sebastianbenz** [comment]
Good find! We should also turn the code snippet into an actual example.

**Comment by philkrie** [comment]
@CrystalOnScript 

Upon further review I'm not entirely sure this is the actual problem and am a little more confused.

In the original example I reported, the input setState on **{radius = event.value}**, so it's not contained in a myCircle object. However, in both macro calls, we referenced **myCircle.radius** on accident. In this case, the amp-bind-macro with "args" instead of "arguments" worked.

When I fixed the input to setState on **{myCircle: {radius = event.value}}**, "arguments" worked, and "args" did not.

I have these cases shown in examples:
setState to **{radius = event.value}**, amp-bind-macro with args works
https://glitch.com/edit/#!/amp-snippets?path=amp-bind-macro-issue_example_1.html
setState to **{radius = event.value}**, amp-bind-macro with args works
setState to **{myCircle: {radius = event.value}}**, amp-bind-macro with arguments works
https://glitch.com/edit/#!/amp-snippets?path=amp-bind-macro-issue_example_2.html

I think what might be happening in the first case is that "args" isn't actually parsed and "arguments" is strictly looking for myCircle.radius. Since we set just radius, "arguments" doesn't pick it up, and the macro with "args" actually just matches radius within the expression, which uses radius. Even though I specify circleArea2(myCircle.radius), it doesn't use the argument or even care that an argument isn't specified. Since radius = x, and the expression is radius * radius * 3.14, it works. If I change radius to something different within the expression, it doesn't work, so I think that's what's happening.

This is 100% my mistake! Should have done some more testing. 

However, I did not know that I could directly reference state from within the amp-bind-macro object w/o using an argument, so perhaps the documentation should clarify this.

I think that a full code sample would clarify usage significantly, this is the only amp-bind-macro example as far as I'm aware of on the site and it is incomplete (myCircle never gets instantiated within the snippet)
